File: frontend/lib/models/EtudiantModel.dart

class EtudiantModel{
  String? id;
  String? nomComplet;
  String? cin;
  String? numInsc;
  String? phone;
  String? email;
  String? photo;
  String? filiere;
  String? classe;

  EtudiantModel({
    this.id,
    this.nomComplet,
    this.email,
    this.phone,
    this.cin,
    this.classe,
    this.filiere,
    this.numInsc,
    this.photo,
});

  Map<String, dynamic> toJson() => {
    'nomComplet': nomComplet,
    "cin":cin,
    "photo":photo,
    "numInsc":numInsc,
    "phone":phone,
    'email': email,
    "filiere":filiere,
    "classe":classe
  };

  EtudiantModel.fromJson(Map<String, dynamic> json)
      : nomComplet = json['nomComplet'] as String,
        cin = json['cin'] as String,
        classe = json['classe'] as String,
        photo = json['photo'] as String,
        email = json['email'] as String,
        phone = json['password'] as String,
        filiere = json['filiere'] as String,
        numInsc = json['numInsc'] as String;

}

File: backend/models/Etudiant.model.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;
const bcrypt = require('bcrypt');

const etudiantSchema = new Schema({
    nomComplet:{
        type:String,
    },
    cin:{
        type:String,
        required:true
    },
    numInsc:{
        type:String,
        required:true,
    },
    email:{
        type:String,
    },
    phone:{
        type:String,
    },
    photo:{
        data:Buffer,
        content:String
    },
    filiere:{
        type:String,
        required:true
    },
    classe:{
        type:String
    }

},{timestamps:true});

etudiantSchema.pre('save',async function(){
    try {
        var etudiant = this;
        const salt = await(bcrypt.genSalt(10));
        const hashpass  =  await bcrypt.hash(etudiant.numInsc,salt);
        etudiant.numInsc=hashpass;
       
    } catch (error) {
        throw error;
    }
});

const Etudiant = mongoose.model('Etudiant',etudiantSchema);

module.exports = Etudiant;